workspaces/tests/types.rs

```diff
@@ -33,6 +35,58 @@ fn test_keypair_secp256k1() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[test]
+fn test_pubkey_serialization() -> anyhow::Result<()> {
+    for key_type in [KeyType::ED25519, KeyType::SECP256K1] {
+        let sk = SecretKey::from_seed(key_type, "test");
+        let pk = sk.public_key();
+        let bytes = pk.try_to_vec()?;
+
+        // Borsh Deserialization should equate to the original public key:
+        assert_eq!(PublicKey::try_from_slice(&bytes)?, pk);
+
+        // invalid public key should error out on deserialization:
+        assert!(PublicKey::try_from_slice(&[0]).is_err());
+    }
+
+    Ok(())
+}
+
+#[test]
+fn test_pubkey_borsh_format_change() -> anyhow::Result<()> {
```

Comment:
Hm. Struggling a bit to understand this test.

Is there a case where this would fail that's directly impacted by the struct here? Otherwise, it looks like a data test is all we're trying to do.

So this would be equivalent, no?

```rust
let mut data = vec![KeyType::ED25519 as u8];
data.extend(bs58::decode("6E8sCci9badyRkXb3JoRpBj5p8C6Tw41ELDZoiihKEtp").into_vec()?);

let new_key = PublicKey::try_from_slice(data.as_slice())?;

assert_eq!(
    new_key.try_to_vec()?,
    bs58::decode("16E8sCci9badyRkXb3JoRpBj5p8C6Tw41ELDZoiihKEtp").into_vec()?
);
```

Reply:
hmm, now that you mention it, yeah it should be equivalent. I wonder why SDK did it like this in the first place

Reply:
oh it's because it had `Vec<u8>.try_into()` but I just got rid of it in favor of borsh

Reply:
Ah, gotcha!